pkg/reftracker/ref_tracker.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package reftracker
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type AppRefTracker struct {
+	lock       sync.Mutex
+	refsToApps map[string]map[string]struct{}
+}
+
+func NewAppRefTracker() *AppRefTracker {
+	refsToApps := map[string]map[string]struct{}{}
+	lock := sync.Mutex{}
+	return &AppRefTracker{refsToApps: refsToApps, lock: lock}
+}
+
+func (a AppRefTracker) AddAppToRefMap(resourceKind, resourceName, namespace, appName string) {
```

Comment:
AddAppToRefMap -> AddAppForRef
GetAppsForRef -> AppsForRef
RemoveRefFromMap -> RemoveRef
RemoveAppFromRefMap -> RemoveApp
CheckAppExistsForRef (remove)

Reply:
All renamed. I instead named `RemoveAppFromRefMap -> RemoveApp` to `RemoveAppFromAllRefs` as it now handles deleting App from all associated refs.